### PR TITLE
[RPC] Transaction details in getblock

### PR DESCRIPTION
**Pull request by salisbury-espinosa**

### CODE REVIEW

**Comment by UdjinM6** [comment]
This could be controlled by yet another rpc parameter or `verbose ` could become 0-2 instead of `bool` to provide all options. And this is exactly what was already implemented in upstream, so I'd rather postpone this till future backports.

**Comment by salisbury-espinosa** [comment]
@UdjinM6 `And this is exactly what was already implemented in upstream, so I'd rather postpone this till future backports.` 
taked from https://github.com/bitcoin/bitcoin/pull/8704/commits/c99ab3ca4b2363b721d6582cf22d5d41896d1037

File: src/rpc/blockchain.cpp [hunk 1]
```diff
@@ -824,8 +824,10 @@ UniValue getblock(const JSONRPCRequest& request)
             "If verbose is true, returns an Object with information about block <hash>.\n"
             "\nArguments:\n"
             "1. \"blockhash\"          (string, required) The block hash\n"
-            "2. verbose                (boolean, optional, default=true) true for a json object, false for the hex encoded data\n"
-            "\nResult (for verbose = true):\n"
+            "2. verbosity              (numeric, optional, default=1) 0 for hex-encoded data, 1 for a json object, and 2 for json object with transaction data\n"
```
[the remark this answers is not in the document]
Comment (salisbury-espinosa) [reply src/rpc/blockchain.cpp]: fixed

File: src/rpc/blockchain.cpp [hunk 2]
```diff
@@ -859,9 +868,14 @@ UniValue getblock(const JSONRPCRequest& request)
     std::string strHash = request.params[0].get_str();
     uint256 hash(uint256S(strHash));
 
-    bool fVerbose = true;
-    if (request.params.size() > 1)
-        fVerbose = request.params[1].get_bool();
+    int verbosity = 1;
+    if (!request.params[1].isNull()) {
```
[the remark this answers is not in the document]
Comment (salisbury-espinosa) [reply src/rpc/blockchain.cpp]: fixed - taked changes from only original PR

**Comment by UdjinM6** [comment]
So, it seems you are fixing the wrong problem :)

You should just add DIP2 (https://github.com/dashpay/dips/blob/master/dip-0002.md) support in your bitcoinjs-lib fork, smth like that https://github.com/UdjinM6/bitcoinjs-lib/commit/69807fd8dd76ed8b7dcb81ccae62e13c667f5cfb

**Comment by salisbury-espinosa** [comment]
> So, it seems like you are fixing the wrong problem :)

The problem is the impossibility of extracting transaction details from the block.
There are 2 ways to solve the problem
1) rpc upgrade to get transaction details by block hash
2) create fork of bitcoinjs-lib with support for all implemented DIP's in dash core

The first method is the most correct, since it relies on the code base where these improvement proposals are implemented ( https://github.com/dashpay/dash ). 

For the second method 
1) it is necessary  to support 2 forks (one for dash and one for the rest)  in the code of the consumer (js parser)  
2) it is necessary to know all the features of the implementation of these DIP's and duplicate this logic in the fork https://github.com/UdjinM6/bitcoinjs-lib/commit/69807fd8dd76ed8b7dcb81ccae62e13c667f5cfb (and therefore, from release to release, it will break and be able to be supported only by the developer who implemented these dip's)

so:

> in your bitcoinjs-lib fork

I would not like to have 2 fork in the consumer code (js)

> You should just add DIP2

it's easy to do only after you have added DIP2  to js ( https://github.com/UdjinM6/bitcoinjs-lib/commit/69807fd8dd76ed8b7dcb81ccae62e13c667f5cfb ) 
:)

**Comment by UdjinM6** [comment]
>The first method is the most correct

Well, that's debatable imo. The output json data is 10-ish times larger than the raw one, so for a full 2mb block you'll have to parse 20mb-ish json. And without patching bitcoinjs-lib you'll have to avoid parsing/producing raw data at all times and would have to only rely on workarounds instead, which throws quite a bit of bitcoinlib-js functionality out of the window imo.

>it is necessary to support 2 forks (one for dash and one for the rest)

Not really. The way I implemented it, it should be backwards compatible with bitcoin/litecoin/etc. so you only need just 1 single fork with these features implemented.

>it is necessary to know all the features of the implementation of these DIP's and duplicate protocol specific features in the fork UdjinM6/bitcoinjs-lib@69807fd (and therefore, from release to release, it will probably break and be able to be supported only by the developer who implemented these dip's)

That's a good point. Maybe we could maintain such a fork in dashpay or dashevo repo. Thoughts @codablock @nmarley ?

**Comment by UdjinM6** [comment]
>what if a litecoin or bitcoin transaction appears with version === 3 and they will need another logic?

True, good point.

**Pull request merged**